src/client/components/table-of-contents/table-of-contents.vue

```diff
@@ -14,5 +15,78 @@
 <script>
 export default {
   props: ['items'],
+  data () {
+    return {
+      currentSection: this.items[0].slug
+    }
+  },
+  mounted () {
+    window.addEventListener('scroll', this.getClosestSection)
```

Comment:
Prefer a "[passive listener](https://developers.google.com/web/updates/2016/06/passive-event-listeners)" or maybe an Intersection Observer (talk to Sanne ;) ).